### PR TITLE
Feature Custom Accounts Grouping

### PR DESCRIPTION
**Pull request by dtrinh50**
https://federal-spending-transparency.atlassian.net/browse/DEV-1059, https://federal-spending-transparency.atlassian.net/browse/DEV-1156

- Implemented the option to group by File C
- Implemented the option to group by Federal accounts
- Created API contract for Custom Accounts

- [x] Code Review

- [x] Design Review

- [x] fedspendingtransparency/usaspending-api#1291 merged

### CODE REVIEW

File: api_contracts/contracts/downloads/CustomAccountDownload.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
++ Request (application/json)
+    + Attributes (object)
+        + account_level: `treasury_account` (required, enum[string])
+            The account level is used to filter for a specific type of file.
+            + Members
+                + treasury_account
+                + federal_account
+        + file_format: `csv` (optional, string)
+        /    The file format that should be returned.
```
Comment (ebdabbs) [inline api_contracts/contracts/downloads/CustomAccountDownload.md]: Extra / character

File: api_contracts/contracts/downloads/CustomAccountDownload.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    Where the file lives in S3
++ message (optional, nullable)
++ status: `finished` (required, string)
++ seconds_elapsed `10.061132` (required, string)
+    
+## FilterObject (object)
++ agency: `all` (optional, string)
+    The agency to filter by
+    + Default: `all`
+    + submission_type: `award_financial` (required, enum[string])
```
Comment (ebdabbs) [inline api_contracts/contracts/downloads/CustomAccountDownload.md]: This line shouldn't be indented.

File: api_contracts/contracts/downloads/CustomAccountDownload.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+## CustomDataResult (object)
++ total_size: 35.055 (required, number)
+    The total size of the file being returned
++ file_name: `012_account_balances_20180613140845.zip` (required, string)
++ total_rows: 652 (required, number)
++ total_columns: 27 (required, number)
++ url: `S3/path_to/bucket/012_account_balances_20180613140845.zip` (required, string)
+    Where the file lives in S3
++ message (optional, nullable)
++ status: `finished` (required, string)
```
Comment (ebdabbs) [inline api_contracts/contracts/downloads/CustomAccountDownload.md]: I believe status can have type `enum[string]` with members `ready`, `running`, `finished`, or `failed`.

File: api_contracts/contracts/downloads/CustomAccountDownload.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    The agency to filter by
+    + Default: `all`
+    + submission_type: `award_financial` (required, enum[string])
+    + Members
+        + account_balances
+        + object_class_program_activity
+        + award_financial
++ fy: `2017` (required, string)
+    The fiscal year to filter by in the format `YYYY`
++ quarter: 1 (required, enum[string])
```
Comment (ebdabbs) [inline api_contracts/contracts/downloads/CustomAccountDownload.md]: I would change this to `1` to ensure it is interpreted as a string

**Comment by AGuyNamedMarco** [comment]
Is it still possible to match the Q selector's colors with the one in our Spending Explorer?
screen shot 2018-06-21 at 11 32 01 am

**Pull request merged**